fiatconnect-api.md

```diff
@@ -383,105 +406,245 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+and returns a `200`.
+
+###### 3.2.1.3.3.1. Success
+
+On success, the server MUST respond with a `200` status code. The server MUST only respond successfully after performing a variety of checks.
+
+If the client did not send session cookies, or if the included cookie does not correspond to a valid session with an expected nonce already attached, the server MUST fail the
+request. A server MUST verify that the signature is valid and corresponds to the `address` field included in the signed message. A server MUST verify that the `nonce` field corresponds
+exactly to the nonce already set on the user's session. A server MUST verify that the `domain` and `uri` fields correspond exactly to the values expected for the
+server's hostname/origin. A server MUST verify that the `Version` and `Chain ID` lines in the signed message are exactly `1` and `42220`, respectively.
+If the `issued-at` field in the signed message is before the server's current timestamp, or if the `issued-at` field is after the `expiration-time` field, the server
+MUST fail the request. If the server's current timestamp is past the `expiration-time` field, the server MUST fail the request. If the `expiration-time` field is
+greater than two weeks (1209600 seconds) ahead of the `issued-at` field, the server MUST fail the request. If all fields are acceptable, and the message
+is formatted correctly, the server MUST authenticate the user's session, and return an HTTP `200`. The server MUST attach the fields from the signed message to the session internally,
+in order to access data about the user and their session in other endpoints.
+
+Once a server authenticates a session using a particular nonce, the server MUST NOT allow any other sessions to be created using that nonce, until `expiration-time` of the session created
+using that nonce has been reached. The implementation of this is up to the server, but will require storing some global state about nonces in order to check if a client is attempting
+to sign-in using a nonce that is temporarily forbidden. Careful consideration should be taken to make sure that nonce management works correctly when scaling server resources.
+
+###### 3.2.1.3.3.2. Failure
+
+On failure, the server MUST return an HTTP `401` error, along with an error code that details exactly why the request failed.
+
+###### 3.2.1.3.3.2.1 `InvalidSignature`
+
+If the signature is invalid or doesn't match the included `address` field, the server MUST return an `InvalidSignature` error.
+
+###### 3.2.1.3.3.2.2 `InvalidParameters`
+
+If the request body is missing, malformed, or fails to pass any of the required checks, the server MUST respond with an `InvalidParameters` error.
+
+#### 3.2.1.4. Using Sessions
+
+Once a client has established a session with a server, they can use that session (by sending the session cookie to the server) to access priveleged endpoints
```

Comment:
What is the exact name of the cookie that the client uses to pass the session?

Reply:
The exact name is unimportant, but a best-effort should probably be made by providers to ensure that the cookie name will not collide with those given by other providers. Will specify this in the spec.